src/shared/libs/iota/quorum.js

```diff
@@ -0,0 +1,415 @@
+                            );
+                        }),
+                ),
+            ),
+        )
+        .then((results) => {
+            const validResults = filter(results, (result) => !isUndefined(result));
+
+            // Get quorum result for latest solid subtangle milestone
+            const latestSolidSubtangleMilestone = determineQuorumResult(
+                map(validResults, (nodeInfo) => nodeInfo.latestSolidSubtangleMilestone),
+            )('latestSolidSubtangleMilestone', QUORUM_THRESHOLD);
+
+            // If nodes cannot agree on the latestSolidSubtangleMilestone
+            // Then just throw an exception
+            if (latestSolidSubtangleMilestone === EMPTY_HASH_TRYTES) {
+                // TODO (laumair): Might need to replace the error message here.
+                throw new Error(Errors.NOT_ENOUGH_HEALTHY_QUORUM_NODES);
+            }
+
+            return latestSolidSubtangleMilestone;
+        });
+};
+
+/**
+ *   Wrapper for quorum enabled iota methods
+ *
+ *   @method Quorum
+ *   @param {array} quorumNodes
+ *
+ *   @returns {object}
+ **/
+export default function Quorum(quorumNodes) {
+    let nodes = uniq(quorumNodes);
+
+    // TODO validate min size of nodes
+    let syncedNodes = [];
+    let lastSyncedAt = new Date();
+
+    const findSyncedNodesIfNecessary = () => {
+        const timeElapsed = (new Date() - lastSyncedAt) / 1000;
+
+        if (isEmpty(syncedNodes) || timeElapsed >= QUORUM_NODES_SYNC_CHECKS_INTERVAL) {
+            return findSyncedNodes(nodes, QUORUM_SIZE, syncedNodes).then((newSyncedNodes) => {
+                syncedNodes = newSyncedNodes;
+                lastSyncedAt = new Date();
+
+                return newSyncedNodes;
+            });
+        }
+
+        return Promise.resolve(syncedNodes);
+    };
+
+    return {
+        /**
+         * Set quorum nodes
+         *
+         * @method setNodes
+         * @param {array} newNodes
+         */
+        setNodes(newNodes) {
+            nodes = union(nodes, uniq(newNodes));
+        },
+        /**
+         * Performs a quorum for wereAddressesSpentFrom api endpoint
+         *
+         * @method wereAddressesSpentFrom
+         * @param {array} addresses
+         *
+         * @returns {Promise}
+         */
+        wereAddressesSpentFrom(addresses) {
+            return findSyncedNodesIfNecessary().then((newSyncedNodes) =>
+                getQuorumForWereAddressesSpentFrom(addresses, newSyncedNodes),
+            );
+        },
+        /**
+         * Performs a quorum for getLatestInclusion api endpoint
+         *
+         * @method getLatestInclusion
+         * @param {array} hashes
+         *
+         * @returns {Promise}
+         */
+        getLatestInclusion(hashes) {
+            return findSyncedNodesIfNecessary().then((newSyncedNodes) =>
+                getQuorumForLatestSolidSubtangleMilestone(newSyncedNodes).then((latestSolidSubtangleMilestone) =>
```

Comment:
this addition is worth mentioning somewhere - in the JSDocs (same for `getBalances`)

Reply:
Did you mean documentation for first getting quorum results for `latestSolidSubtangleMilestone`? Have updated [JSDocs](https://github.com/laumair/trinity-wallet/blob/feat/quorum/src/shared/libs/iota/quorum.js#L345-L347) for this method.